pkg/metrics/scraper.go

```diff
@@ -10,32 +10,39 @@ import (
 	"github.com/prometheus/common/model"
 )
 
+type DataPoint struct {
```

Comment:
SamplePair can be used here? https://github.com/prometheus/common/blob/e27922a100211b71acc8e65327b61d5fe22e2a77/model/value_float.go#L69

Reply:
Didn't want to expose p8s client data structures in the scraper interface. As this is a generic interface that can be extended for non promql datasources in future.